cms/models.py

```diff
@@ -62,6 +62,45 @@ def get_context(self, request):
         return context
 
 
+class FaqsPage(Page):
+    """
+    CMS page for questions
+    """
+    content_panels = Page.content_panels + [
+        InlinePanel('faqs', label='Frequently Asked Questions'),
+    ]
+
+    def get_context(self, request):
+        js_settings = {
+            "gaTrackingID": settings.GA_TRACKING_ID,
+            "host": webpack_dev_server_host(request),
+            "programId": self.get_parent().program.id,
+            "faculty": faculty_for_carousel(self.faculty_members.all()),
+        }
+        username = get_social_username(request.user)
+        context = super(FaqsPage, self).get_context(request)
+
+        courses_info = []
+        for course in self.get_parent().program.course_set.all().order_by(
+                'position_in_program'
+        ):
+            courses_info.append((course, get_course_enrollment_text(course), get_course_url(course)))
+
+        context["style_src"] = get_bundle_url(request, "style.js")
+        context["public_src"] = get_bundle_url(request, "public.js")
+        context["style_public_src"] = get_bundle_url(request, "style_public.js")
+        context["authenticated"] = not request.user.is_anonymous()
+        context["signup_dialog_src"] = get_bundle_url(request, "signup_dialog.js")
+        context["faculty_carousel_src"] = get_bundle_url(request, "faculty_carousel.js")
+        context["username"] = username
+        context["js_settings_json"] = json.dumps(js_settings)
+        context["title"] = self.get_parent().title
+        context["courses_info"] = courses_info
+        context['programpage'] = self.get_parent().first()
```

Comment:
There's a lot of copy and paste here from the ProgramPage. Is it all necessary? If it is, can we refactor to keep it DRY?